tests/prompts_test.py

```diff
@@ -0,0 +1,362 @@
+                    completion = client.chat.completions.create(
+                        model=QWEN_MODEL,
+                        temperature=0,
+                        max_tokens=1000,
+                        messages=[
+                            {
+                                "role": "system",
+                                "content": QWEN_PROMPT_TEMPLATE.format(
+                                    OPTIONS=OPTIONS,
+                                    INSTRUCTION=INSTRUCTION,
+                                ),
+                            },
+                            {
+                                "role": "user",
+                                "content": specification,
+                            },
+                        ],
+                    )
+
+                    llm_response = completion.choices[0].message.content
+                    index_of_result_word = llm_response.find("RESULT")
+
+                    for sentence in str(llm_response).split("\n"):
+                        if "RESULT" in sentence:
+                            if expected_result in llm_response[index_of_result_word:]:
+                                match_count += 1
+                                print(
+                                    f"Circle-{index_of_circle + 1} "
+                                    f"QWEN: Test-{index_of_testcase} PASSED"
+                                )
+                            else:
+                                failed_testcases += (
+                                    f"\nCircle-{index_of_circle + 1} "
+                                    f"Testcase-{index_of_testcase}: "
+                                    f"Expected: {expected_result} "
+                                    f"LLM: {sentence}\n\n"
+                                )
+                                print(
+                                    f"Circle-{index_of_circle + 1} "
+                                    f"QWEN: Test-{index_of_testcase} FAILED: "
+                                    f"Result: {sentence}"
+                                )
+                            result_testing_tuples.append(
+                                (
+                                    index_of_circle + 1,
+                                    index_of_testcase,
+                                    expected_result,
+                                    sentence,
+                                    llm_response,
+                                    QWEN_PROMPT_TEMPLATE,
+                                    INSTRUCTION,
+                                    OPTIONS,
+                                    QWEN_MODEL,
+                                )
+                            )
+
+        self.save_response_to_csv(
+            result_testing_tuples, "qwen"
+        )  # Save prompts and llm responses to csv file
+
+        accuracy = round(match_count / (len(self.testcases) * RETRY_COUNT), 2)
+        print(f"Accuracy is {accuracy}")
```

Comment:
should be extracted into separate function

Reply:
Done